lib/msf/core/post/file.rb

```diff
@@ -704,6 +704,61 @@ def _unix_max_line_length
     line_max
   end
 
+  def search( root=nil, glob='*.*', recurse=true, timeout=-1 )
+    matches = []
+
+    if session.type == 'meterpreter'
+      return session.fs.file.search(root, glob, recurse, timeout)
+
+    elsif session.type == 'powershell'
+      list = cmd_exec("Get-ChildItem #{recurse ? '-Recurse': ''} -Path #{root}| where {! $_.PSIsContainer} | Format-Table Name, Length, Directory").split("\n")
+      list.each do |file_info|
+        file_info = file_info.split
+        attrib = {}
+        attrib['name'] = file_info[0]
+        next unless attrib['name'].match? (glob)
+        attrib['size'] = file_info[1]
```

Comment:
Doesn't this need to be converted to an integer?

Reply:
yes, will change that :+1: